src/probability_theory/martingale.lean

```diff
@@ -307,4 +316,49 @@ end
 
 end submartingale
 
+section nat
+
+variables {𝒢 : filtration ℕ m0} [sigma_finite_filtration μ 𝒢]
+
+namespace submartingale
+
+lemma integrable_stopped_value {f : ℕ → α → ℝ} (hf : submartingale f 𝒢 μ) {τ : α → ℕ}
```

Comment:
I'm not sure how to generalize the second lemma since it requires `set_integral_mono_ae` which applies only to real-valued functions. Perhaps `set_integral_mono_ae` should be generalized but I'm not sure if the integral is monotone in a more general context.

Reply:
The monotonicity of the integral should be generalizable to something like a `normed_lattice_add_comm_group`. I am proving a similar statement in #10714 (under review). `ℝ` is fine for now then.